api/v1alpha3/machinehealthcheck_types.go

```diff
@@ -93,6 +93,11 @@ type MachineHealthCheckStatus struct {
 	// +kubebuilder:validation:Minimum=0
 	CurrentHealthy int32 `json:"currentHealthy,omitempty"`
 
+	// RemediationsAllowed is the number of further remediations allowed by this machine health check before
+	// maxUnhealthy short circuiting will be applied
+	// +kubebuilder:validation:Minimum=0
+	RemediationsAllowed int32 `json:"remediationsAllowed,omitempty"`
```

Comment:
This was not added by this PR. If it is deleted in the main branch, we shouldn't be adding this back.
Will check more to see how this is deleted in the main branch.

Reply:
It was never there..it will be added anyway by #3884

Reply:
I see, then I think it'd be better to wait it merged first.